tidy3d/web/api/mode.py

```diff
@@ -137,6 +145,97 @@ def run(
+    verbose : bool
+        If True, displays a progress bar. If False, runs silently.
+    max_workers : int
+        Maximum number of concurrent workers to use for processing the batch of simulations.
+    max_retries : int
+        Maximum number of retries for each simulation in case of failure before giving up.
+    retry_delay : int
+        Delay in seconds between retries when a simulation fails.
+    progress_callback_upload : Callable[[float], None], optional
+        Optional callback function called when uploading file with ``bytes_in_chunk`` as argument.
+    progress_callback_download : Callable[[float], None], optional
+        Optional callback function called when downloading file with ``bytes_in_chunk`` as argument.
+
+
+    Returns
+    -------
+    List[ModeSolverData]
+        A list of ModeSolverData objects containing the results from each simulation in the batch. None is placed in the list for simulations that fail after all retries.
+    """
+    console = get_logging_console()
+
+    # Check type to make sure the user is submitting a list of mode solvers
+    if False in [type(x)== ModeSolver for x in mode_solvers]: 
+        console.print("[red]ERROR: Stopped running a batch of mode solver simulations. Please provide a list of mode solvers of type [cyan]`td.plugins.mode.mode_solver.ModeSolver` [red]to run a batch of mode solver simulations.")
+        return
+
+    # Result files where the data is stored after downloading
+    if results_files is None:
+        results_files = [f"{folder_name}/mode_solver_results_{i}.hdf5" for i in range(len(mode_solvers))]
+
+    results = [None] * len(mode_solvers)
+    
+    # Handling simulation to make sure there are more than one tries if simulation produces some error
+    def handle_simulation(index, retries=0):
+        try:
+            # We'll create a separate progress bar for batch simulations so we'll set verbose=False here for individual runs
+            result = run(mode_solvers[index], f"{task_name}_{index}", f"mode_solver_batch_{index}", folder_name, results_files[index], False, progress_callback_upload, progress_callback_download)
+            results[index] = result
+        except Exception as exc:
+            if retries < max_retries:
+                # Wait before retrying
+                time.sleep(retry_delay)  
+                handle_simulation(index, retries + 1)
+            else:
+                results[index] = None 
+
+    # Using multithreading to run mode solver simulations concurrently
+    with ThreadPoolExecutor(max_workers=max_workers) as executor:
+        futures = [executor.submit(handle_simulation, i) for i in range(len(mode_solvers))]
+        # Creating a progress bar for batch simulations using rich progress
+        if verbose:
+            with Progress(console=console) as progress:
+                pbar_message = f"[cyan]Running a batch of [deep_pink4]{len(mode_solvers)} mode solvers [cyan]in parallel.\n"
```

Comment:
some purists might take issue with "in parallel" and could email us for clarification. Maybe we should just leave this part out and just saying that we're running a batch of {N} mode solvers should be enough